internal/ir/infra.go

```diff
@@ -0,0 +1,65 @@
+package ir
+
+import (
+	cfgv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
+)
+
+// Infra defines managed infrastructure.
+type Infra struct {
+	// Proxy defines managed proxy infrastructure.
+	Proxy *ProxyInfra
+}
+
+// ProxyInfra defines managed proxy infrastructure.
+type ProxyInfra struct {
+	// TODO: Figure out how to represent metadata in the IR.
+	// xref: https://github.com/envoyproxy/gateway/issues/173
+	//
+	// Name is the name used for managed proxy infrastructure.
+	Name string
+	// Namespace is the namespace used for managed proxy infrastructure.
+	Namespace string
+	// OwnerRefs is a list of objects used to generate this object.
+	OwnerRefs []ObjectReference
```

Comment:
curious why this field is needed ?

Reply:
So we know the object, e.g. Gateway, that was used to create the `ProxyInfra`. The owning object can be inferred, especially if namespaced-scoped proxy infra resources are required to be in the same namespace as the Gateway.

Reply:
how does the Infra Manager plan on using this information, imho the IR layer removes an hard association with the Gateway API layer, which is done by the Gateway API / Resource Translator, any informational bits should live in the Metadata field.

Reply:
> how does the Infra Manager plan on using this information
The infra mgr can use `OwnerRef` to add a label to the infra object that describes the owning object, e.g. Gateway ns/name.

Reply:
ah very similar use case to how I was planning on leveraging the `Metadata` field in the Xds IR to embed the owner ref into the Xds config metadata fields, we should either pick the OwnerRef or the Metadata approach and keep it consistent across the IRs or solve it with https://github.com/envoyproxy/gateway/issues/173

Reply:
I think the use case is a little different:

OwnerRef is a Kubernetes construct that tells Kubernetes that an object should be cleaned up when its owner is removed.

Metadata on IR objects is about recording where certain config came from.

I'd agree that we should make any Kubernetes proxy config we generate have an OwnerRef to something, maybe the Envoy Gateway deployment (or whatever's running the infra manager), so that if that gets removed, the data paths associated with it get cleaned up, and that this is a useful place to store that info.

@danehans  has noted on L15 that we need to put the metadata info into here once we figure out what it is.

Reply:
I'll remove `OwnerRefs` and we'll figure it out as part of the [metadata issue](https://github.com/envoyproxy/gateway/issues/173).